### DISCUSSION TITLE
Ngnix config file should be moved to the root of the repo

### DISCUSSION THREAD

**Discussion by ido777** [danny-avila/LibreChat#2086] [Feature Requests & Suggestions]
### What happened?

The nginx configuration file located at  ./client/nginx.conf should be moved to either 
./nginx.conf 
./nginx/nginx.conf 

For 4 reasons 
- it is not part of the client code 
- people might use different web server than nginx 
- it is not intuitive to search nginx files within react files.
- given the fact that recreating the client docker will only update the Nginx config file but not the client code it is extremely confusing. 

(BTW A similar issue but less problematic is with  ./mkdocs.yml that a better place might be 
./docs/mkdocs.yml  
or 
./docs/src/mkdocs.yml

along with requirments.txt  and read.MD file that will explain how to serve the documents (e.g. mkdocs serve -f ../../mkdocs.yml )

### Steps to Reproduce

Nginx is a different entity than client entity and even if there were packed in the same docker(which is not the case today) still their source should be separated to avoid confusion.  

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fuegovic** [comment]
Not sure what you mean here, aren't you already talking about this in #2087 ?

**Comment by ido777** [comment]
@fuegovic you are correct that my original post was not clear enough.
I will explain here, the difference between this and #2087 and later I will edit the original post here to make this issue better understood.

Those are 2 different issues 
This issue is about the source code and the project structure.
The other issue is about packing different applications in the docker than their name. 
Is it clear enough now?

**Reply by fuegovic** [reply]
What do you mean by the nginx code? Are you suggesting that the nginx **configuration** file should be stored in a different location?

**Reply by ido777** [reply]
Exactly - Great minds think alike ;-) , 
I just finished editing the original post and then I realized you suggested the same.

**Reply by fuegovic** [reply]
This is diverging from your original post, but since you now talk about it in the edit: 

regarding mkdocs.yml, I believe it must be stored in the specified location to be recognized (if my memory serves me correctly)

**Reply by ido777** [reply]
it has the option of -f similar to docker so you can do 
`mkdocs serve` -> will search the specified location  (IMO current folder)  
or  
`mkdocs serve -f ../../mkdocs.yml` -> or any other location will look for the file mentioned

**Reply by ido777** [reply]
Apologies for not being clear in the first place, thanks for correcting me.
